### ISSUE TITLE
uncaught global is not defined error, when trying to use interface/example object in external project, how to use just the models?

### ISSUE THREAD

**Issue by michaelbats** [szokodiakos/typegoose#352]
Hi, I've been trying to use my exported models within my angular project -- this is strictly the types and some examples objects, the code compiles fine but then immediatelly throws 
```
Uncaught (in promise): ReferenceError: global is not defined
ReferenceError: global is not defined
    at Object../node_modules/bson/lib/bson/map.js (map.js:4)
    at __webpack_require__ (bootstrap:83)
    at Object../node_modules/bson/lib/bson/bson.js (bson.js:3)
    at __webpack_require__ (bootstrap:83)
    at Object../node_modules/mongoose/lib/drivers/browser/binary.js (binary.js:8)
    at __webpack_require__ (bootstrap:83)
    at Object../node_modules/mongoose/lib/drivers/browser/index.js (index.js:7)
    at __webpack_require__ (bootstrap:83)
    at Object../node_modules/mongoose/lib/browser.js (browser.js:5)
    at __webpack_require__ (bootstrap:83)
```
which seems to be due to Mongoose, although I don't need anything from mongoose/typegoose, the TS interface/type (which doesn't exist at runtime anyway) and the example model which doesn't use typegoose, just fills in some basic values as in the example below

My doc example:
```
export type IJob = Without<JobSchema, 'getModelForClass' | 'setModelForClass' | 'buildSchema'>;

export const JobDocExample: IJob = {
	_id: 'string',
	title: 'string',
	reference: 12345,
	createdAt: new Date(Date.now()).toISOString()
};

```

Is this a bug or am I using this wrong? how would it be possible to get that example doc to work without mongoose?

**Comment by hasezoey** [comment]
from the error, it seems like this is not an typegoose error, and it is not officially supported in a browser

**Comment by michaelbats** [comment]
@hasezoey  it seemed to actually need a polyfill -- but the error was only when trying to import stuff from that package. The fixes were angular-specific though. I added to my polyfills: 
```
(window as any).global = window;
// @ts-ignore
window.Buffer = window.Buffer || require('buffer').Buffer;
```

Now I'm a tad bit concerned about how the bundle size will be affected by using it like this, I can't seem to be able to use the models + examples without installing typegoose + mongoose + @types/mongoose as I don't really/fully understand what happens at runtime, I'll keep investigating but my basic assumption -- since types don't exist at runtime, it shouldn't have any effect.

**Comment by hasezoey** [comment]
this can be closed, because it dosnt seems like a typegoose issue & it is not officially supported to work in the browser / without node

@Ben305 

if any news on this, please reply, thanks

**Comment by michaelbats** [comment]
@hasezoey I agree there's no issue with typegoose -- but I was wondering if its worth looking at a way to export those types/examples without the need for mongoose/typegoose as a dependency?

Because all the types and objects have their interface based on a class that contains the @prop() and other decorators, attempting to use these in another app seems to need typegoose + mongoose to be present and then they throw out (legitimate) errors due to being used in something like an angular app.

Currently getting ``` ReferenceError: process is not defined ``` at runtime from mongoose in angular, tracked it down all the way to the @prop decorator in the class.

**Comment by hasezoey** [comment]
@michaelbats you could still define everything in a different class(or in an interface if not needed at runtime) and just extend(/implement) it in the class along with typegoose

**Comment by michaelbats** [comment]
@hasezoey  so I went a bit further down the rabbit hole, just to give you some context: 

 - this is a lazy loaded module, the app works and its fine, the errors are thrown out when I load that specific lazy loaded module, it hasn't got anything to it.
- if I remove the references to stuff I'm using from that package, it loads and all is fine. 

I just literally tested and all I had imported from that package was enums! nothing special, just enums. if I have those exact enums locally, everything is fine, if I try to import them from the package, I get the errors above. 

Starting to think that this might be something about how I export my own module maybe?

Importing the types/interfaces themselves and all is ok, using the package with mongoose to talk to the DB all is fine as well. Weird thing.
